packages/sdk/src/domain/services/CurrencyService.ts

```diff
@@ -0,0 +1,54 @@
+import { Currency } from "@heliofi/common";
+import type { HelioApiAdapter } from "../../infrastructure";
+
+export class CurrencyService {
+  private currencies!: Currency[];
+
+  constructor(private apiService: HelioApiAdapter) {}
+
+  setCurrencies(currencies: Currency[]) {
+    this.currencies = currencies;
+  }
+
+  getCurrencyBySymbol(symbol: string): Currency {
+    if (!this.hasCurrencyResult()) {
+      throw new Error(`You should call currencyService.getCurrencies() before`);
+    }
+
+    const currency = this.currencies?.find(
+      (currencyItem) => currencyItem.symbol === symbol
+    );
+    if (currency == null) {
+      throw new Error(`Unable to find currency: ${currency}`);
```

Comment:
throwing an error might be a bit too harsh on SDK, is it possible to return Currency | undefined ? or does that require lots of changes